src/main/java/org/opensearch/security/privileges/PitAccessEvaluator.java

```diff
@@ -0,0 +1,344 @@
+
+        // Skip pit evaluation for "NodesGetAllPITs" action, since it fetches all PITs across the cluster
+        // for privilege evaluation
+        if(action.startsWith("cluster:admin")) {
+            return presponse;
+        }
+        try {
+            if (request instanceof GetAllPitNodesRequest) {
+                if (((GetAllPitNodesRequest) request).getGetAllPitNodesResponse() != null) {
+                    return presponse;
+                }
+                return handleGetAllPitsAccess(request, clusterService, user, securityRoles,
+                        action, resolver, presponse);
+            } else if (request instanceof DeletePitRequest) {
+                DeletePitRequest deletePitRequest = (DeletePitRequest) request;
+                List<String> pitIds = deletePitRequest.getPitIds();
+                if (pitIds.size() == 1 && "_all".equals(pitIds.get(0))) {
+                    return handleDeleteAllPitAccess(deletePitRequest, clusterService, user, securityRoles,
+                            action, resolver, presponse);
+                } else {
+                    return handleExplicitPitsAccess(deletePitRequest.getPitIds(), clusterService, user, securityRoles,
+                            action, resolver, presponse);
+                }
+            } else if (request instanceof PitSegmentsRequest) {
+                PitSegmentsRequest pitSegmentsRequest = (PitSegmentsRequest) request;
+                List<String> pitIds = pitSegmentsRequest.getPitIds();
+                if (pitIds.size() == 1 && "_all".equals(pitIds.get(0))) {
+                    return handleGetAllPitSegmentsAccess(pitSegmentsRequest, clusterService, user, securityRoles,
+                            action, resolver, presponse);
+                } else {
+                    return handleExplicitPitsAccess(pitSegmentsRequest.getPitIds(), clusterService, user, securityRoles,
+                            action, resolver, presponse);
+                }
+            }
+        } catch(InterruptedException e) {
+            Thread.currentThread().interrupt();
+            log.error(e.toString());
+        }
+        return presponse;
+    }
+
+    /**
+     * Handle access for Get All PITs access
+     */
+    private PrivilegesEvaluatorResponse handleGetAllPitsAccess(final ActionRequest request, final ClusterService clusterService,
+                                        final User user, SecurityRoles securityRoles, final String action,
+                                        IndexNameExpressionResolver resolver,
+                                        PrivilegesEvaluatorResponse presponse) throws InterruptedException {
+        List<ListPitInfo> pitInfos = getAllPitInfos((GetAllPitNodesRequest) request);
+        // if cluster has no PITs, then allow the operation to pass with empty response
+        if(pitInfos.isEmpty()) {
+            presponse.allowed = true;
```

Comment:
The authorization on a request should not be decided by whether there are PITs in a cluster. It should always be evaluated with permissions instead. If the user does not have right permissions, the request should be denied no matter whether there are PITs in a cluster or how many PITs there are.

Reply:
If the user does not have permissions, we do throw exception. I'll add some ITs which can assert the same.

Now this logic makes sure that we don't throw permission exception when there are simply no PITs created in the cluster.

Reply:
>Now this logic makes sure that we don't throw permission exception when there are simply no PITs created in the cluster.

If a user does not have PIT search permissions, even if there are not PITs in the cluster, the request still should be denied. As I said, authorizing a request or not should only be decided by the permissions the user has(privilege evaluator).

Reply:
Addressed the comment, we're denying if there are no pits so resolving this and similar ones

Reply:
so as long as the user does not have PITs , we throw 403. The behavior is consistent and the above does not happen.

User might not have PITs because :
- PITs are present in the cluster but the user does not have access
- PITs are not present in the cluster